static/scripts/rewards/web3/erc721-permit.ts

```diff
@@ -31,9 +31,9 @@ export function claimErc721PermitHandler(reward: Erc721Permit) {
 
     buttonController.showLoader();
     try {
-      const nftContract = new ethers.Contract(reward.permit.permitted.token, nftRewardAbi, signer);
+      const nftContract = new ethers.Contract(reward.tokenAddress, nftRewardAbi, signer);
 
-      const tx: TransactionResponse = await nftContract.safeMint(reward.request, reward.signature);
+      const tx: TransactionResponse = await nftContract.safeMint(reward, reward.signature);
```

Comment:
I think it's not correct to pass our permit struct. `safeMint` has its own struct defined in the [ABI](https://github.com/ubiquity/pay.ubq.fi/blob/development/static/scripts/rewards/abis/nft-reward-abi.ts)

Reply:
can you clarify what you mean? @whilefoo

Reply:
@whilefoo thank you, nice catch. Didn't see the signature was simply any kind of object, so TS was not complaining. Fixed the payload.

Reply:
it would be nice to have TS types for contracts

Reply:
can be addresses in subsequently PRs with typechain